cloudant-sync-datastore-core/src/test/java/com/cloudant/sync/internal/mazha/CouchConfig.java

```diff
@@ -34,33 +39,98 @@
     // a reverse proxy
     private URI rootUri;
 
+    private String username;
+
+    private String password;
+
     private List<HttpConnectionRequestInterceptor> requestInterceptors = new ArrayList
             <HttpConnectionRequestInterceptor>();
     private List<HttpConnectionResponseInterceptor> responseInterceptors = new ArrayList
             <HttpConnectionResponseInterceptor>();
 
     public CouchConfig(URI rootUri) {
-        this.rootUri = rootUri;
+        this(rootUri,
+                Collections.<HttpConnectionRequestInterceptor>emptyList(),
+                Collections.<HttpConnectionResponseInterceptor>emptyList(),
+                null,
+                null);
     }
 
     public CouchConfig(URI rootUri,
                        List<HttpConnectionRequestInterceptor> requestInterceptors,
-                       List<HttpConnectionResponseInterceptor> responseInterceptors) {
+                       List<HttpConnectionResponseInterceptor> responseInterceptors,
+                       String username,
+                       String password) {
         this.rootUri = rootUri;
         this.requestInterceptors = requestInterceptors;
         this.responseInterceptors = responseInterceptors;
+        this.username = username;
+        this.password = password;
+
     }
 
-    public List<HttpConnectionRequestInterceptor> getRequestInterceptors() {
+
+    public List<HttpConnectionRequestInterceptor> getRequestInterceptors(boolean includeCookie) {
+        CookieInterceptor cookieInterceptor = buildCookieInterceptor();
+        if (includeCookie && cookieInterceptor != null) {
+            List<HttpConnectionRequestInterceptor> requestInterceptors = new ArrayList
+                    <HttpConnectionRequestInterceptor>();
+            requestInterceptors.addAll(this.requestInterceptors);
+            requestInterceptors.add(cookieInterceptor);
+            return requestInterceptors;
+        }
         return requestInterceptors;
     }
 
+    public List<HttpConnectionResponseInterceptor> getResponseInterceptors(boolean includeCookie) {
+        CookieInterceptor cookieInterceptor = buildCookieInterceptor();
+        if (includeCookie && cookieInterceptor != null) {
+            List<HttpConnectionResponseInterceptor> requestInterceptors = new ArrayList
+                    <HttpConnectionResponseInterceptor>();
+            requestInterceptors.addAll(this.responseInterceptors);
+            requestInterceptors.add(cookieInterceptor);
+            return requestInterceptors;
+        }
+        return responseInterceptors;
+    }
+
+    private CookieInterceptor buildCookieInterceptor() {
```

Comment:
f9d6d74c5f898f40e9ae679176659aef9dca00d7 worked for me locally.
I feel like either I must be missing some nuance or you're over thinking it. Or perhaps it will fail in the CI and demonstrate to me why that approach doesn't work!